aclocal.m4

```diff
@@ -314,9 +280,85 @@ AC_DEFUN([MP_CONFIG_TARBALL], [
 		fi
 		AC_MSG_NOTICE([=== finished configuring in $ac_dir ($mp_popdir/$ac_dir)])
 	fi
+
 	cd "$mp_popdir"
 ])
 
+dnl Configure a project contained in a .tar.gz, .tgz or .tar.bz2 tarball,
+dnl extracting it previously, if necessary. Different from AC_CONFIG_SUBDIRS
+dnl (on which this macro is based), you can pass parameters to the
+dnl sub-configure script.
+dnl
+dnl Parameters:
+dnl  - The relative path to the tarball
+dnl  - The relative path to the directory that will be extracted from the
+dnl    tarball and contains the configure script to be run (can be a sub directory of the extracted directory)
+dnl  - Parameters to pass to the configure script
+dnl
+dnl MP_CONFIG_TARBALL([path-to-tarball], [dir-extracted-from-tarball-with-configure], [configure-parameters])
+AC_DEFUN([MP_CONFIG_TARBALL], [
+	dnl Warning: Don't use $GZIP and $BZIP2 here, both tools interpret these
+	dnl env variables as additional parameters.
+	AC_PATH_PROG(GZIP_BIN, [gzip], [])
+	AC_PATH_PROG(BZIP2_BIN, [bzip2], [])
+
+	mp_tarball="$1"
+	ac_dir=$2
+
+	if ! test -d "$ac_dir"; then
+		mp_tarball_vendordir="$(dirname "$mp_tarball")"
+		AS_MKDIR_P(["$mp_tarball_vendordir"])
+		AC_MSG_NOTICE([=== extracting $mp_tarball])
+		mp_tarball_extract_cmd=
+		case "$mp_tarball" in
+			*.tar.gz | *.tgz)
+				if test "x$GZIP_BIN" = "x"; then
+					AC_MSG_ERROR([gzip not found])
+				fi
+				mp_tarball_extract_cmd="$GZIP_BIN"
+				;;
+			*.tar.bz2 | *.tbz2)
+				if test "x$BZIP2_BIN" = "x"; then
+					AC_MSG_ERROR([bzip2 not found])
+				fi
+				mp_tarball_extract_cmd="$BZIP2_BIN"
+				;;
+			*)
+				AC_MSG_ERROR([Don't know how to extract tarball $mp_tarball])
+				;;
+		esac
+		(cd "$mp_tarball_vendordir"; umask 0022; "$mp_tarball_extract_cmd" -d < "$ac_abs_confdir/$mp_tarball" | tar xf - || AC_MSG_ERROR([failed to extract $mp_tarball]))
+	fi
+	if ! test -d "$ac_dir"; then
+		AC_MSG_ERROR([tarball $mp_tarball did not extract to $ac_dir])
+	fi
+
+	MP_CONFIG_SUBDIR(["$ac_dir"])
```

Comment:
That was just a mistake. Interestingly, it still ended up in the right directory in the vendor-destroot.